src/hip_device.cpp

```diff
@@ -35,16 +35,15 @@ hipError_t hipGetDevice(int* deviceId) {
     hipError_t e = hipSuccess;
 
     auto ctx = ihipGetTlsDefaultCtx();
+    int devid = 0;
 
-    if (deviceId != nullptr) {
-        if (ctx == nullptr) {
-            e = hipErrorInvalidDevice;  // TODO, check error code.
-            *deviceId = -1;
-        } else {
-            *deviceId = ctx->getDevice()->_deviceId;
-        }
+    if (ctx == nullptr) {
+        devid = -1;
     } else {
-        e = hipErrorInvalidValue;
+        devid = ctx->getDevice()->_deviceId;
+    }
+    if (deviceId != nullptr) {
```

Comment:
You can use else part to report 'hipErrorInvalidValue' which was there in the older implementation.